pageserver/src/tenant/init_dir.rs

```diff
@@ -0,0 +1,132 @@
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    path::{Path, PathBuf},
+};
+
+use anyhow::Context;
+
+use tracing::warn;
+use utils::id::TimelineId;
+
+use crate::is_uninit_mark;
+
+fn timeline_id_from_path(path: &Path) -> anyhow::Result<TimelineId> {
+    path.file_stem()
+        .and_then(OsStr::to_str)
+        .unwrap_or_default()
+        .parse::<TimelineId>()
+        .with_context(|| {
+            format!(
+                "Could not parse timeline id out of the path {}",
+                path.display()
+            )
+        })
+}
+
+pub enum Decision {
+    Load,
+    RemoveTemporary(PathBuf),
+    RemoveUninitMark(PathBuf),
+}
+
+#[derive(Default)]
+pub struct TimelinesToLoad {
```

Comment:
```suggestion
pub(super) struct TimelinesToLoad {
```

Reply:
Probably other types should be `pub(super)` instead of `pub`. Just better defaults.